mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/ElasticSearchIntegrationTest.java

```diff
@@ -104,4 +122,73 @@ protected void initializeMailboxManager() throws Exception {
         storeMailboxManager.setMessageSearchIndex(messageSearchIndex);
         storeMailboxManager.init();
     }
+
+    @Ignore("MAILBOX-301 Lucene terms length exceeded on some emails")
+    @Test
+    public void termsBetweenElasticSearchAndLuceneLimitDueTuNonAsciiCharsShouldNotMakeIndexingFail() throws Exception {
```

Comment:
Ok after analyzing the mapping, when I read that:
```
.startObject(TEXT_BODY)
    .field(TYPE, STRING)
    .startObject(FIELDS)
        .startObject(RAW)
            .field(TYPE, STRING)
            .field(ANALYZER, CASE_INSENSITIVE)
            .field(IGNORE_ABOVE, MAXIMUM_TERM_LENGTH)
        .endObject()
    .endObject()
.endObject()
```
I think about you PR and wonder, how could it work. Because here the mapping of the TEXT_BODY is saying we are going to analyze the field two time and then create to two field. One is named TEXT_BODY and the other one is named TEXT_BODY.RAW.
The thing is that in the mapping, you define the ignore_above only for the raw version! So if you index a mail with a too long term in the the textBody we should have an error.
I put a break point in the builder of indexableMessage when running one of your test and I noticied that the resulting message have a value for text but not for bodyText neither bodyHtml.
Therefore, your test are not complete and neither your fix IMO